word_guess.rb

```diff
@@ -0,0 +1,269 @@
+    if word_done?
+      win_game
+    end
+    if !more_tries?
+      lose_game
+    end
+  end
+
+  # this method actually gets the input
+  def get_input
+    puts "What letter do you guess?"
+    @guess = gets.chomp.upcase
+
+    # this checks if you guessed the right word.
+    while @guess.length > 1
+      if @guess == @word.upcase
+        win_game
+      else
+        #second time we use this- refactor into a method?
+        @wrong_count -= 1
+        puts "You have made a terrible guess!"
+        puts "How could you pick that letter?!"
+        @used_letters.push(@guess)
+        update_board
+        puts "You have #{ @wrong_count } guesses left."
+        @guess = gets.chomp.upcase
+      end
+    end
+
+    # this checks if the guess is a single character and if that character is a letter!
+    while @guess.length != 1 || @alphabet_array.include?(@guess) == false
+        puts "Your guess is unusable, please guess again!"
+        @guess = gets.chomp.upcase
+    end
+    # echoes guess to user
+    puts "Your guess was #{ @guess }."
+  end
+
+  # this checks if the letter is in the word, and if you already guessed it.
+  def check_letter
+
+    unless @used_letters.include?(@guess) == false
+        puts "You have already guessed this letter!"
+        puts "Please guess again!"
+        @guess = gets.chomp.upcase
+    else
+        # add letter to used letters array
+        @used_letters.push(@guess)
+    end
+
+    if @word.upcase.include?(@guess)
+      for index in 0..@word.length - 1
+        if @word[index].upcase == @guess
+          # update_word #took it out of method just to check variable
+          @word_so_far[index] = @guess
+        end
+      end
+      puts "What a great guess!"
+      puts "Good job!"
+      update_board
+    else
+      #guess is wrong
+      @wrong_count -= 1
+      puts "You have made a terrible guess!"
+      puts "How could you pick that letter?!"
+      update_board
+      puts "You have #{ @wrong_count } guesses left."
+    end
+  end
+
+  # this checks if you have more wrong guesses available
+  def more_tries?
+      @wrong_count >= 1
+  end
+
+  def word_done?
+    @word_so_far == @word.upcase
+  end
+
+  # this prints the winning game board
+  def win_game
+    #call board drawing
+    @wrong_count = 6
+    update_board
+    message = "YAY! YOU WIN! Your prize is an "
+    message2 = "#{@good_prize}"
+    puts message.colorize(:yellow) + message2.colorize(:magenta)
+    system "say", message
+    system "say", message2
+    exit
+  end
+
+  # this prints the losing game board
+  def lose_game
+    #call board drawing
+    wrong_count = 0
+    # update_board
+    message = "OH, NO! YOU LOSE!! Your prize is #{@bad_prize}..."
+    puts message.colorize(:red)
+    puts "The correct word was: #{@word.colorize(:yellow)}"
+    system "say", message
+    exit
+  end
+
+  # this prints the game boards and updates them.
+  def update_board
+    ghost_one = "=======".colorize(:cyan)
```

Comment:
Rather than storing each of these pieces in individual variables, could you put each piece into an array and use an array index to access each piece?